nx/lib/nx.ex

```diff
@@ -685,7 +685,15 @@ defmodule Nx do
     assert_keys!(opts, [:type, :names, :backend])
     shape = Nx.shape(tensor_or_shape)
     names = Nx.Shape.named_axes!(opts[:names] || names!(tensor_or_shape), shape)
-    type = Nx.Type.normalize!(opts[:type] || Nx.Type.infer(max - min))
+    range_type = Nx.Type.infer(max - min)
```

Comment:
`infer` will always infer the 64-bit version of whatever the output type of `min - max` is. If I do

```elixir
Nx.random_uniform({5, 5, 5}, -10000, 10000, type: {:s, 8})
```
It will work, but I will end up with truncated values. I'm not really sure if letting this behavior in is ideal or something we should prevent altogether. I guess since we allow the same sort of truncation in `Nx.as_type` it might not matter that much.

Reply:
Yes, we generally document that given high values will overflow/underflow. We can add more checks in the future though, it doesn't hurt.